BASE/Test/Microsoft.ApplicationInsights.Test/Microsoft.ApplicationInsights.Tests/Channel/RedirectHttpHandlerTests.cs

```diff
@@ -0,0 +1,193 @@
+            using var localServer2 = LocalInProcHttpServer.MakeTargetServer(url: LocalUrl2, response: helloString);
+
+            var client = new MyCustomClient(url: LocalUrl1, new RedirectHttpHandler());
+
+            // Default behavior. 1st server will redirect to 2nd.
+            var testStr1 = await client.GetAsync();
+            Assert.AreEqual(helloString, testStr1);
+            Assert.AreEqual(1, localServer1.RequestCounter);
+            Assert.AreEqual(1, localServer2.RequestCounter);
+
+            // Redirect is cached. Request will go to 2nd server.
+            var testStr2 = await client.GetAsync();
+            Assert.AreEqual(helloString, testStr2);
+            Assert.AreEqual(1, localServer1.RequestCounter, "redirect should be cached");
+            Assert.AreEqual(2, localServer2.RequestCounter);
+
+            // wait for cache to expire
+            await Task.Delay(shortCache * 2);
+
+            // Default behavior. 1st server will redirect to 2nd.
+            var testStr3 = await client.GetAsync();
+            Assert.AreEqual(helloString, testStr3);
+            Assert.AreEqual(2, localServer1.RequestCounter);
+            Assert.AreEqual(3, localServer2.RequestCounter);
+        }
+
+        /// <summary>
+        /// Verify behavior of HttpClient with <see cref="RedirectHttpHandler"/>.
+        /// In this test, server1 will redirect to itself.
+        /// Verify that <see cref="RedirectHttpHandler.MaxRedirect"/> is enforced.
+        /// </summary>
+        [TestMethod]
+        public async Task VerifyMaxRedirects()
+        {
+            using var localServer1 = LocalInProcHttpServer.MakeRedirectServer(url: LocalUrl1, redirectUrl: LocalUrl1, cache: TimeSpan.FromDays(1));
+
+            var client = new MyCustomClient(url: LocalUrl1, new RedirectHttpHandler());
+
+            var testStr1 = await client.GetAsync();
+            Assert.AreEqual("redirect", testStr1);
+
+            Assert.AreEqual(RedirectHttpHandler.MaxRedirect + 1, localServer1.RequestCounter, $"expecting 1 original request + {RedirectHttpHandler.MaxRedirect} additional requests");
+        }
+
+        /// <summary>
+        /// This class is a wrapper around <see cref="HttpClient"/>.
+        /// I'm using this to simplify the tests above.
+        /// </summary>
+        private class MyCustomClient
+        {
+            public readonly Uri uri;
+            public readonly HttpClient httpClient;
+
+            public MyCustomClient(string url, RedirectHttpHandler httpMessageHandler = null)
+            {
+                this.uri = new Uri(url);
+                this.httpClient = (httpMessageHandler == null)
+                    ? new HttpClient()
+                    : new HttpClient(httpMessageHandler);
+            }
+
+            public async Task<string> GetAsync()
+            {
+                var result = await this.httpClient.SendAsync(new HttpRequestMessage(HttpMethod.Get, this.uri));
```

Comment:
This is all very well, but Breeze uses POST (see Transmission.CreateRequestMessage). Don't you think we should test that too?
In particular, Breeze's use of cache-control on a POST is non-standard.

Also, the point of writing our own RedirectHttpHandler is to ensure that auth headers are preserved on redirect. We should have a test for that too.